components/hooks/useMigration.ts

```diff
@@ -278,11 +278,15 @@ export const useMigration = (): UseMigrationReturn => {
   /**
    * Clear synchronization data and restart the synchronization process
    */
-  const restartSynchronization = useCallback(() => {
+  const restartSynchronization = useCallback(async () => {
     // Clear synchronization data
     ledgerState$.clearSynchronization()
-    // Restart synchronization process
-    ledgerState$.synchronizeAccounts()
+
+    const result = await ledgerState$.connectLedger()
+    if (result.connected && result.isAppOpen) {
+      ledgerState$.synchronizeAccounts()
+    }
```

Comment:
If the app is not connected or the app is not opened, shouldn't we display some error? By doing this, the restartSynchornization will do nothing on those cases. Do you think that it is correct?